src/binder/bind_node_visitor.cpp

```diff
@@ -440,6 +441,43 @@ void BindNodeVisitor::Visit(common::ManagedPointer<parser::InsertStatement> node
           }
 
           ins_val->Accept(common::ManagedPointer(this).CastManagedPointerTo<SqlNodeVisitor>());
+
+          // if we have constant values of variable types being inserted into fixed size columns,
+          // we need to resolve that
+          if (ins_val->GetExpressionType() == parser::ExpressionType::VALUE_CONSTANT) {
+            auto const_val = ins_val.CastManagedPointerTo<parser::ConstantValueExpression>();
+            bool is_variable = const_val->GetReturnValueType() == type::TypeId::VARCHAR ||
+                               const_val->GetReturnValueType() == type::TypeId::VARBINARY;
```

Comment:
Does it make sense to apply this logic to a `VARBINARY`?

Reply:
yeah although you need to throw a different error in the `VARBINARY` case and not truncate